src/app/Plans/components/Wizard/FilterVMsForm.tsx

```diff
@@ -87,17 +85,35 @@ const FilterVMsForm: React.FunctionComponent<IFilterVMsFormProps> = ({
   ]);
 
   const getNodeBadgeContent = (node: InventoryTree, isRootNode: boolean) => {
+    if (!treeQuery.data) return null;
     const { treeType } = form.values;
     const { isItemSelected, selectedItems } = treeSelection;
-    const selectedDescendants = flattenInventoryTreeNodes(node).filter(isItemSelected);
-    const numVMs = getAvailableVMs(selectedDescendants, vmsQuery.data || [], treeType).length;
+    const allDescendants = isRootNode
+      ? treeQuery.data.flattenedNodes
+      : treeQuery.data.getDescendants(node, true);
+    const selectedDescendants = allDescendants.filter(isItemSelected);
+    const numVMs = getAvailableVMs(
+      treeQuery.data,
+      selectedDescendants,
+      vmsQuery.data,
+      treeType
+    ).length;
     const rootNodeSuffix = ` VM${numVMs !== 1 ? 's' : ''}`;
     if (numVMs || isItemSelected(node) || (isRootNode && selectedItems.length > 0)) {
       return `${numVMs}${isRootNode ? rootNodeSuffix : ''}`;
```

Comment:
Doesn't this represent the number of VMs selected from _descendents_ of this node in the tree? At first I thought it was a summation of all VMs.. Maybe we can make the badge text more consistent in some way? Maybe we should use a Label component here instead of Badge since it's not only a numeric count inside it.

Reply:
That is a fantastic point, and something we should follow up on. Opened an issue: https://github.com/konveyor/forklift-ui/issues/718